hack/prepare-for-integration-tests.sh

```diff
@@ -396,6 +429,38 @@ else
   popd >/dev/null
 fi
 
+#
+# Call a post-install script
+# simplifies passing the $tag which may be necessary if the current local build is to be
+# referenced, for example, deploying via a Carvel package rather than our ytt mechanism
+# running it after the above also allows appending to the environment variable file
+if [ "$post_install" != "undefined" ] ; then
+  log_note "The post-install script will be called with $tag..."
+  $post_install post-install-script $tag
+fi
+
+#
+# Test user for the authenticator
+# the authenticator may be deployed in alternative ways (ex. carvel package) but regardless we need a test user.
+#
+log_note "Creating test user for local-user-authenticator..."
+test_username="test-username"
+test_groups="test-group-0,test-group-1"
+test_password="$(openssl rand -hex 16)"
+
+kubectl create secret generic "$test_username" \
+  --namespace local-user-authenticator \
+  --from-literal=groups="$test_groups" \
+  --from-literal=passwordHash="$(htpasswd -nbBC 10 x "$test_password" | sed -e "s/^x://")" \
+  --dry-run=client \
+  --output yaml |
+  kubectl apply -f -
+
+#
+# Regardless of how the local-user-authenticator is installed, we need the webhook bundle in the environment file.
+#
```

Comment:
If we chunk out the Carvel script differently, this can potentially return to where it was.